vortex-datafusion/src/persistent/cache.rs

```diff
@@ -10,16 +10,17 @@ use vortex_array::aliases::DefaultHashBuilder;
 use vortex_array::stats::{Precision, Stat};
 use vortex_dtype::DType;
 use vortex_error::{VortexError, VortexResult, vortex_err};
-use vortex_file::{Footer, SegmentSpec, VortexOpenOptions};
-use vortex_io::ObjectStoreReadAt;
+use vortex_file::{Footer, SegmentSpec, VortexFile, VortexOpenOptions};
 use vortex_layout::LayoutRegistry;
 use vortex_layout::segments::SegmentId;
+use vortex_metrics::VortexMetrics;
 
-#[derive(Debug, Clone)]
-pub(crate) struct FooterCache {
-    inner: Cache<Key, Footer, DefaultHashBuilder>,
+#[derive(Clone)]
+pub(crate) struct VortexFileCache {
+    inner: Cache<Key, VortexFile, DefaultHashBuilder>,
```

Comment:
Seems like we store the footer twice, once in `VortexFile` and another in the `VortexFileIO` implementation.